### PR TITLE
fix: save chart in dashboard

### PR DESCRIPTION
**Pull request by rephus**
Closes: #9817

### Description:

[Screencast from 18-04-24 17:21:46.webm](https://github.com/lightdash/lightdash/assets/1983672/fe2542fa-48ca-4c3d-812f-e001ba8c77e5)

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/services/DashboardService/DashboardService.ts [hunk 1]
```diff
@@ -532,7 +532,7 @@ export class DashboardService extends BaseService {
                 {
                     tiles: dashboard.tiles,
                     filters: dashboard.filters,
-                    tabs: dashboard.tabs,
+                    tabs: dashboard.tabs || [],
```
Comment (rephus) [inline packages/backend/src/services/DashboardService/DashboardService.ts:535]: not mandatory for this fix, as it is also patched in the frontend, but good to keep

File: packages/frontend/src/components/common/modal/ChartCreateModal/SaveToSpaceOrDashboard.tsx [hunk 2]
```diff
@@ -312,15 +313,14 @@ export const SaveToSpaceOrDashboard: FC<SaveToSpaceOrDashboardProps> = ({
                     },
                     ...getDefaultChartTileSize(savedData.chartConfig?.type),
                 };
-                await updateDashboard({
-                    name: values.dashboardName
-                        ? values.dashboardName
-                        : 'untitled',
+                const updateFields: DashboardVersionedFields = {
                     filters: selectedDashboard.filters,
```
Comment (owlas) [inline packages/frontend/src/components/common/modal/ChartCreateModal/SaveToSpaceOrDashboard.tsx:317]: this doesn't include `name` anymore, don't we need it?

Comment (rephus) [reply packages/frontend/src/components/common/modal/ChartCreateModal/SaveToSpaceOrDashboard.tsx:317]: No, we don't.  That was also part of the issue , we were overriding the name with untitled.

**Review by ZeRego — APPROVED** [review]

**Comment by rephus** [comment]
Unrelated ee2e errors

**Pull request merged**